### PR TITLE
Use JQuery.detach on removeItem so custom data attributes are retained

### PR DESCRIPTION
**Pull request by john3parker**
I'm using a custom Item renderer with some custom data attribute fields. When .remove is used, JQuery removes the data-* and event handlers. Since selectize caches the rendered item it doesn't get created each time so the data-* attributes and even handlers are lost. To prevent his situation I changed the .remove() to .detach(). I also added a test to check for data-* attribute retention.

### CODE REVIEW

**Comment by giero** [comment]
It's kind of connected with my issue: https://github.com/selectize/selectize.js/pull/1315
